src/main/java/com/multiplex/booking/BookingController.java

```diff
@@ -0,0 +1,279 @@
+        @ApiResponse(responseCode = "400", description = "Invalid json supplied", 
+            content = @Content)})
+    ReservationResponse replaceSeat(@Valid @RequestBody List<Seat> newSeats) {
+
+        Showing showing = showings.findById(newSeats.get(0).getShowingId())
+        .orElseThrow(() -> new ShowingNotFoundException(newSeats.get(0).getShowingId()));
+
+        // check time, reservation should be before the movie
+        Duration duration = Duration.between(LocalDateTime.now(),showing.getShowingTime());
+        long minutes = duration.toMinutes();
+
+        if(minutes <= 15) throw new CannotReserveException();
+
+        Set<Seat> seats = showing.getSeats();
+
+        double total = 0;
+
+        for (Seat newSeat : newSeats) {
+            Optional<Seat> seatOptional = seats.stream()
+                .filter(seat -> seat.getId().equals(newSeat.getId()))
+                .findFirst();
+
+            if (seatOptional.isPresent()) {
+                Seat seat = seatOptional.get();
+
+                String name = newSeat.getName();
+                String surname = newSeat.getSurname();
+                String ticketType = newSeat.getTicketType();
+
+                // seat cannot be reserved, name, surname and ticket type is required
+                if(seat.isReserved()) throw new WrongInputException();
+                if(name == null) throw new WrongInputException();
+                if(surname == null) throw new WrongInputException();
+                if(ticketType == null) throw new WrongInputException();
+
+                // name must start with an uppercase
+                if(!Character.isUpperCase(name.charAt(0))) throw new WrongInputException();
+
+                // surname must start with an uppercase and could consist of two parts separated with
+                // a single dash, in this case the second part should also start with a capital letter
+                if(surname.contains("-")){
+                    String[] surname_parts = surname.split("-");
+                    String first_surname = surname_parts[0];
+                    String second_surname = surname_parts[1];
+                    if(!Character.isUpperCase(first_surname.charAt(0))) throw new WrongInputException();
+                    if(!Character.isUpperCase(second_surname.charAt(0))) throw new WrongInputException();
+                }
+                else{
+                    if(!Character.isUpperCase(surname.charAt(0))) throw new WrongInputException();
+                }
+
+                seat.setReserved(true);
+                seat.setName(name);
+                seat.setSurname(surname);
+                seat.setTicketType(ticketType);
+
+                if(newSeat.getTicketType().equals("adult")) total += ADULT_PRICE;
+                if(newSeat.getTicketType().equals("student")) total += STUDENT_PRICE;
+                if(newSeat.getTicketType().equals("child")) total += CHILD_PRICE;
```

Comment:
Czy potrzebujemy sprawdzać wszystkie 3 warunki? Czy poza if'ami moglibyśmy inaczej zapisać takie sprawdzenie?

Reply:
Jednym z rozwiązań byłoby użycie switcha:
```suggestion
switch (newSeat.getTicketType()) {
    case "adult":
        total += ADULT_PRICE;
        break;
    case "student":
        total += STUDENT_PRICE;
        break;
    case "child":
        total += CHILD_PRICE;
        break;
    default:
        throw new WrongInputException();
        break;
}
```
Jeśli zostałaby podana niepoprawna nazwa typu biletu, zwróciłabym błąd. Mogłabym również użyć konstrukcji z else-if:
```suggestion
                if (newSeat.getTicketType().equals("adult")) {
                     total += ADULT_PRICE;
                } else if (newSeat.getTicketType().equals("student")) {
                     total += STUDENT_PRICE;
                } else if (newSeat.getTicketType().equals("child")) {
                     total += CHILD_PRICE;
                } else  {
                     throw new WrongInputException();
                }
```